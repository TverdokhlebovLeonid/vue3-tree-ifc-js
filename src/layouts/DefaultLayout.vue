<script setup lang="ts">
import { computed } from 'vue'
import { useRoute } from 'vue-router'

const route = useRoute()
const title = computed((): string => (route?.meta?.title as string) || '')
</script>

<template>
  <div class="default-layout">
    <el-container>
      <el-header>
        <el-text>{{ title }}</el-text>
      </el-header>
      <el-main>
        <slot />
      </el-main>
    </el-container>
  </div>
</template>

<style scoped lang="sass">
.default-layout
  background-color: $color-bg-light
  height: 100VH
  border-radius: 5px
  .el-header
    margin-bottom: 20px
    display: flex
    background-color: $color-bg-brand
    border-radius: 5px
    .el-text
      color: $color-text-light
</style>
