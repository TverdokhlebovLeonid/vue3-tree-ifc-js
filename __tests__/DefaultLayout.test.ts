import { mount } from '@vue/test-utils'
import DefaultLayout from '@/layouts/DefaultLayout.vue'
import type { ComponentWrapperType } from './interfaceIfcTests'
import { mockSlot, mockRoute } from './dataMock'

describe('DefaultLayout test component', () => {
  let component: ComponentWrapperType<InstanceType<typeof DefaultLayout>>

  beforeEach(() => {
    vi.mock('vue-router', () => ({
      useRoute: () => mockRoute,
    }))

    component = mount(DefaultLayout, mockSlot)
  })

  it('Testing the slot', () => {
    expect(component.find('.default-layout').find('.vitest-slot-div').find('p').text()).toBe(
      'Vitest Slot',
    )
  })

  it('Snapshot component', () => {
    expect(component.html()).toMatchSnapshot()
  })

  it('Testing the route', () => {
    expect(
      component
        .find('.default-layout')
        .find('.el-container')
        .find('.el-header')
        .find('span')
        .text(),
    ).toBe('Просмотр IFC файлов - Vitest')
  })
})
