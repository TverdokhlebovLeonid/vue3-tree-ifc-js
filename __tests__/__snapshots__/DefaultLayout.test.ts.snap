// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DefaultLayout test component > Snapshot component 1`] = `
"<div data-v-75d0be1e="" class="default-layout">
  <section data-v-75d0be1e="" class="el-container is-vertical">
    <header data-v-75d0be1e="" class="el-header"><span data-v-75d0be1e="" class="el-text">Просмотр IFC файлов - Vitest</span></header>
    <main data-v-75d0be1e="" class="el-main">
      <div class="vitest-slot-div">
        <p>Vitest Slot</p>
      </div>
    </main>
  </section>
</div>"
`;
